pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -760,7 +762,10 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 			CommonTlsContext: &auth.CommonTlsContext{},
 			Sni:              tls.Sni,
 		}
-
+		// Use service accounts in service if TLS settings does not have subject alt names for service entries.
+		if opts.serviceRegistry == string(serviceregistry.External) && len(tls.SubjectAltNames) == 0 {
```

Comment:
TBH, just feels very weird it has to be aware of the `serviceregistry` when building cluster tls context.

Reply:
That is because of the API we had in service entry - If it is not there we did not have to use it - And we can not set service accounts as subject alt names in all cases (for example user specified TLS etc...). Do you have a better suggestion to handle this?

Reply:
Thinking about this - I can think of couple of options
- https://github.com/istio/api/blob/master/networking/v1alpha3/destination_rule.proto#L961 - instead of override if we append the service entry SANs and DR SANs it might work and we do not need registry check. This is already being done for workload selector based WLE https://github.com/istio/api/blob/master/networking/v1alpha3/service_entry.proto#L953

- Add a specific field in SubjectAltNames which would be populated from ServiceEntry and use it here i.e. do not reuse ServiceAccounts for SubjectAltNames